docs/source/algorithms/CreateCacheFilename-v1.rst

```diff
@@ -9,50 +9,52 @@
-to use from the property manager (empty is use all), and 
-a string array (or List) of other properties to use, and 
-a directory for cache files to exist in (default described below).
-
-The list of property names will be used to select which of the properties 
-in the PropertyManager will be used to calculate the hash 
-and will be interpreted as globbing.
-
-The string array of other_properties will be key/value pairs of properties 
-that should be considered, but are not in the provided PropertyManager.
-
-If a directory is not specified, cache files will go into a cache 
-subdirectory of ConfigService::getUserPropertiesDir().
-On unix this will be ~/.mantid/cache.
-
-The algorithm will convert all properties to strings as 
-"%s=%s" % (property.name, property.valueAsStr), sort the list, 
+The purpose of this algorithm is to create a unique filename for a
+cache so that a workflow can reuse results from previous computations.
+
+The algorithm will accept a prefix,
+:class:`~mantid.kernel.PropertyManager`, list of properties to use
+from the property manager (empty is use all), and a string array (or
+List) of other properties to use, and a directory for cache files to
+exist in (default described below).
+
+The list of property names will be used to select which of the
+properties in the :class:`~mantid.kernel.PropertyManager` will be used
+to calculate the hash and will be interpreted as globbing.
+
+The string array of other_properties will be key/value pairs of
+properties that should be considered, but are not in the provided
+:class:`~mantid.kernel.PropertyManager`.
+
+If a directory is not specified, cache files will go into a cache
+subdirectory of :class:`ConfigService.getUserPropertiesDir()
+<mantid.kernel.ConfigServiceImpl.getUserPropertiesDir>`.  On unix this
+will be ``~/.mantid/cache``.
+
+The algorithm will convert all properties to strings as
+``"%s=%s" % (property.name, property.valueAsStr)``, sort the list,
 then convert it to a sha1.
 
-A filename with the form <location>/<prefix>_<sha1>.nxs 
-will be returned as the output property.
-If no prefix is specified then file result will be <location>/<sha1>.nxs.
-
-* property_manager: an instance of PropertyManager from which property values
-  can be retrieved. None means we don't care about property manager 
-  -- all properties will come from other_properties
-* properties: a list of strings. each string is a property managed by the 
-  given property_manager, or it can be glob pattern to match prop
-  names too. but empty list means taking all properties 
-  from the property_manager
-* other_properties: a list of strings. each string is in the form of
-  "key=value" for one property not managed by the property_manager.
+A filename with the form ``<location>/<prefix>_<sha1>.nxs`` will be
+returned as the output property.  If no prefix is specified then file
+result will be ``<location>/<sha1>.nxs``.
+
+* ``property_manager``: an instance of
```

Comment:
Are the items in this list the properties of the algorithm? In that case, the underscores could be changed to CamelCase, i.e. `property_manager` could be `PropertyManager` and so on.